spring-integration/spring-boot-autoconfigure/src/main/java/ai/timefold/solver/spring/boot/autoconfigure/util/PojoInliner.java

```diff
@@ -0,0 +1,419 @@
+package ai.timefold.solver.spring.boot.autoconfigure.util;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.RecordComponent;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.IdentityHashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import ai.timefold.solver.core.impl.domain.common.ReflectionHelper;
+
+import org.apache.commons.text.StringEscapeUtils;
+import org.springframework.javapoet.CodeBlock;
+import org.springframework.javapoet.TypeSpec;
+
+public final class PojoInliner {
```

Comment:
Agreed, but sadly using XML serialization causes GraalVM to complain about a JAR file being in the native heap, even when the XML serialization is only used for writing the String (outside the generated class with no parsing!)

Reply:
I see your point. Still, this doesn't change my mind - I am not yet convinced we have exhausted all possibilities. 

Please go to the Spring guys, explain what you need to do, and ask them what they think would be the proper solution. If they say that this is the best we can possibly do, then I have no choice but to accept it.

I suggest you ask Geoffrey, he can introduce you to some people.

Reply:
Well, apparently registering too many things for reflection is a bad thing. GraalVM did not like when `java.lang.Class` was registered for reflection, since it made it think its cachedHashCode field was reachable. Not registering java.lang.Class and java.lang.ClassLoader does make it work (although JAXB accesses itself using reflection, so I opted for Jackson instead)